hudi-common/src/main/java/org/apache/hudi/BaseHoodieTableFileIndex.java

```diff
@@ -203,6 +203,9 @@ public int getFileSlicesCount() {
 
   @Override
   public void close() throws Exception {
+    if (cachedAllInputFileSlices instanceof ExternalSpillableMap) {
```

Comment:
@yihua I wasn't able to confirm if this close is actually being called from the unit test for hoodie spark data sources on the read side. Is it a known gap ? 

https://github.com/apache/hudi/blob/master/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieHadoopFsRelationFactory.scala#L260C3-L260C11

Reply:
Spark manages the file index object for relations, and if Spark did not call `close` we need to see how we can eagerly release resources.